src/main/java/com/salesforce/dynamodbv2/mt/mappers/sharedtable/impl/MtAmazonDynamoDbBySharedTable.java

```diff
@@ -414,11 +415,21 @@ public GetItemResult getItem(GetItemRequest getItemRequest) {
         return getItemResult;
     }
 
-    TableMapping getTableMapping(String virtualTableName) {
+    Optional<TableMapping> getTableMapping(String virtualTableName) {
         try {
-            return tableMappingCache.get(virtualTableName, () ->
-                tableMappingFactory.getTableMapping(
-                    new DynamoTableDescriptionImpl(mtTableDescriptionRepo.getTableDescription(virtualTableName))));
+            return tableMappingCache.get(virtualTableName, () -> {
+                try {
+
+                    return Optional.of(tableMappingFactory.getTableMapping(
+                        new DynamoTableDescriptionImpl(mtTableDescriptionRepo.getTableDescription(virtualTableName))));
+                } catch (ResourceNotFoundException e) {
+                    // This isn't great, but we're assuming a missing virtual table entry here is a deleted virtual
+                    // table, and thus, shouldn't be mapping records to MtRecords. Instead return null, but really,
+                    // we should make sure we let deleted records flow through a stream and expire out, before removing
+                    // the virtual table entry.
+                    return Optional.empty();
```

Comment:
since there seem to be a lot of `get()` calls on the return value of `getTableMapping` without `isPresent()` checks, isn't this trading `ResourceNotFoundException`s (possibly with more information) for `NoSuchElementException`s (when this catch block is reached)?

Reply:
yes it is trading RNFE with NSEE, but in scenarios we don't expect to hit that condition, ie: from web request processing of normal CRUD. The only scenario we expect an RNFE is following a delete of a virtual table, and a subsequent stream trying to process data from that deleted virtual table